src/QueryPipeline/Pipe.cpp

```diff
@@ -394,11 +400,19 @@ void Pipe::addExtremesSource(ProcessorPtr source)
 
     assertBlocksHaveEqualStructure(header, source_header, "Pipes");
 
-    if (collected_processors)
-        collected_processors->emplace_back(source);
-
     extremes_port = &source->getOutputs().front();
-    processors->emplace_back(std::move(source));
+
+    addProcessor(std::move(source));
+}
+
+void Pipe::activatePartialResult(UInt64 partial_result_limit_, UInt64 partial_result_duration_ms_)
+{
+    if (!is_partial_result_active)
+        partial_result_ports.assign(output_ports.size(), nullptr);
```

Comment:
Do we expect `activatePartialResult` called once and `partial_result_ports` is empty? Consider adding a comment or/and `assert` to make this conditions clearer